### PR TITLE
[hlm-833]: decompress file from filestore

### PR DESCRIPTION
**Pull request by shivab-egov**
1. Created GZIP decompressor to decompress gzip files. 
2. Added test cases for Decompressor.

### CODE REVIEW

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/compressionService/Compression.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.digit.health.sync.service.compressionService;
+
+import org.digit.health.sync.service.compressionService.compressor.Compressor;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class Compression {
```
Comment (jayantp-egov) [inline digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/compressionService/Compression.java:9]: Remove the compressionService directory. Rename the class to CompressionService

**Review by jayantp-egov — COMMENTED** [review]
Please make requested changes.

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/compressionService/Compression.java (hunk 1 above)
Comment (shivab-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/compressionService/Compression.java]: Done

[the remark this answers is not in the document]
Comment (shivab-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/compressionService/Compression.java]: Done

File: digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/compressor/GzipCompressor.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.digit.health.sync.service.compressor;
+
+import java.io.*;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.ZipException;
+
```
[the remark this answers is not in the document]
Comment (shivab-egov) [reply digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/compressor/GzipCompressor.java:8]: Done

**Review by jayantp-egov — APPROVED** [review]

**Pull request merged**